addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/handler/KM200ThingHandler.java

```diff
@@ -0,0 +1,439 @@
+            gateway.refreshChannel(channel);
+        } else {
+            logger.warn("Unsupported Command: {} Class: {}", command.toFullString(), command.getClass());
+        }
+    }
+
+    /**
+     * Choose a tag for a channel
+     *
+     */
+    Set<String> checkTag(String unitOfMeasure, Boolean readOnly) {
+        Set<String> tags = new HashSet<String>();
+        if (unitOfMeasure.indexOf("°C") == 0 || unitOfMeasure.indexOf("K") == 0) {
+            if (readOnly) {
+                tags.add("CurrentTemperature");
+            } else {
+                tags.add("TargetTemperature");
+            }
+        }
+        return tags;
+    }
+
+    /**
+     * Choose a category for a channel
+     *
+     */
+    String checkCategory(String unitOfMeasure, String topCategory, Boolean readOnly) {
+        String category = null;
+        if (unitOfMeasure.indexOf("°C") == 0 || unitOfMeasure.indexOf("K") == 0) {
+            if (unitOfMeasure.indexOf("°C") == 0 || unitOfMeasure.indexOf("K") == 0) {
+                if (readOnly) {
+                    category = "Temperature";
+                } else {
+                    category = "Heating";
+                }
+            }
+        } else if (unitOfMeasure.indexOf("kW") == 0 || unitOfMeasure.indexOf("kWh") == 0) {
+            category = "Energy";
+        } else if (unitOfMeasure.indexOf("l/min") == 0 || unitOfMeasure.indexOf("l/h") == 0) {
+            category = "Flow";
+        } else if (unitOfMeasure.indexOf("Pascal") == 0 || unitOfMeasure.indexOf("bar") == 0) {
+            category = "Pressure";
+        } else if (unitOfMeasure.indexOf("rpm") == 0) {
+            category = "Flow";
+        } else if (unitOfMeasure.indexOf("mins") == 0 || unitOfMeasure.indexOf("minutes") == 0) {
+            category = "Time";
+        } else if (unitOfMeasure.indexOf("kg/l") == 0) {
+            category = "Oil";
+        } else if (unitOfMeasure.indexOf("%%") == 0) {
+            category = "Number";
+        } else {
+            category = topCategory;
+        }
+        return category;
+    }
+
+    /**
+     * Creates a new channel
+     *
+     */
+    Channel createChannel(ChannelTypeUID channelTypeUID, ChannelUID channelUID, String root, String type,
+            String currentPathName, String description, String label, Boolean addProperties, Boolean switchProgram,
+            StateDescription state, String unitOfMeasure) {
+        Channel newChannel = null;
+        Map<String, String> chProperties = new HashMap<>(1);
+        String itemType = "";
+        String category = null;
+        if (type.compareTo("Number") == 0) {
+            itemType = "NumberType";
+            category = "Number";
+        } else if (type.compareTo("String") == 0) {
+            itemType = "StringType";
+            category = "Text";
+        }
+        ChannelType channelType = new ChannelType(channelTypeUID, false, itemType, ChannelKind.STATE, label,
```

Comment:
Please check out the new ChannelTypeBuilder: `ChannelTypeBuilder.state(...).with...`

Note: this API is very new and not yet available on the latest Eclipse SmartHome stable build. If you don#t want to wait for it just leave the code as it is.

Reply:
Hi,
ok, I removed the unused channel-definitions. I will leave this code here as is.
Markus